### PR TITLE
refactor(connector): [CryptoPay] Remove Default Case Handling

### PR DESCRIPTION
**Pull request by ashutuptiwari**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Removed the default match case as mentioned in the issue

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by ashutuptiwari** [comment]
Please review @swangi-kumari and let me know of any changes that are to be made.

**Comment by ashutuptiwari** [comment]
On it @SanchithHegde

**Comment by ashutuptiwari** [comment]
@SanchithHegde I'm unable to run this locally, please check if the changes i made are working as expected. If yes, I'll revisit the formatting issues and that should be it.

**Comment by SanchithHegde** [comment]
@ashutuptiwari Yeah looks like compilation would be successful, formatting should be the only one remaining.

**Comment by ashutuptiwari** [comment]
Alright then @SanchithHegde

**Comment by ashutuptiwari** [comment]
Should work just fine @SanchithHegde

**Comment by ashutuptiwari** [comment]
Please review

**Comment by ashutuptiwari** [comment]
Alright, I'll loo into it @swangi-kumari

**Comment by ashutuptiwari** [comment]
Please approve the workflow @SanchithHegde

File: crates/router/src/connector/cryptopay/transformers.rs [hunk 1]
```diff
@@ -69,9 +69,23 @@ impl TryFrom<&CryptopayRouterData<&types::PaymentsAuthorizeRouterData>>
                     custom_id: item.router_data.connector_request_reference_id.clone(),
                 })
             }
-            _ => Err(errors::ConnectorError::NotImplemented(
-                "payment method".to_string(),
-            )),
+            api_models::payments::PaymentMethodData::CardRedirect(_)
```
Comment (prasunna09) [inline crates/router/src/connector/cryptopay/transformers.rs:72]:
```suggestion
            api_models::payments::PaymentMethodData::Card(_)
            | api_models::payments::PaymentMethodData::CardRedirect(_)
```

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/cryptopay/transformers.rs [hunk 2]
```diff
@@ -69,9 +69,25 @@ impl TryFrom<&CryptopayRouterData<&types::PaymentsAuthorizeRouterData>>
...
-            )),
+            api_models::payments::PaymentMethodData::Card(_)
+            | api_models::payments::PaymentMethodData::CardRedirect(_)
+            | api_models::payments::PaymentMethodData::Wallet(_)
+            | api_models::payments::PaymentMethodData::PayLater(_)
+            | api_models::payments::PaymentMethodData::BankRedirect(_)
+            | api_models::payments::PaymentMethodData::BankDebit(_)
+            | api_models::payments::PaymentMethodData::BankTransfer(_)
+            | api_models::payments::PaymentMethodData::Card(_)
+            | api_models::payments::PaymentMethodData::Crypto(_)
```
Comment (prasunna09) [inline crates/router/src/connector/cryptopay/transformers.rs:80]:
```suggestion
```
Please remove these two match arms, as these were already covered.

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Comment by ashutuptiwari** [comment]
please review @prasunna09

**Review by prasunna09 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @ashutuptiwari!

**Review by Sangamesh26 — APPROVED** [review]
LGTM :)

**Comment by ashutuptiwari** [comment]
Happy Happy Happy

**Comment by ashutuptiwari** [comment]
Thank you @SanchithHegde @prasunna09 @swangi-kumari for your inputs on this. Learnt a lot throught this contribution. Will definitely look forward to making more contributions. 
Cheers!

**Comment by swangi-kumari** [comment]
Hey @ashutuptiwari , 
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Comment by ashutuptiwari** [comment]
Hi guys, I see my name isn't on the contributors list, nor have I received the swag. 
Is this a mistake or is it amid the delivery process?
Thank you
@SanchithHegde @swangi-kumari @Sangamesh26 @prasunna09

**Comment by VedantKhairnar** [comment]
Hey @ashutuptiwari 
I'll have a word with the vendor around the shipment status.
Will get back to you soon. Thanks for your patience.

**Comment by ashutuptiwari** [comment]
Alright thanks @VedantKhairnar

**Pull request merged**